app/controllers/requests_controller.rb

```diff
@@ -199,4 +205,51 @@ def split_requests(requests)
     @pending_requests = requests.select(&:pending?)
     @resolved_requests = requests.reject(&:pending?)
   end
+
+  def enough_resources
+    def get_num_cpu(host)
```

Comment:
Maybe move this method (and the two below) to `api/v_sphere/host` so it can be called more conveniently? There they could also be tested properly :)